cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/LeaderReplicaDistributionGoal.java

```diff
@@ -0,0 +1,354 @@
+                                                   Set<Goal> optimizedGoals,
+                                                   OptimizationOptions optimizationOptions) {
+    PriorityQueue<Broker> eligibleBrokers = new PriorityQueue<>((b1, b2) -> {
+      int result = Integer.compare(b2.leaderReplicas().size(), b1.leaderReplicas().size());
+      return result == 0 ? Integer.compare(b1.id(), b2.id()) : result;
+    });
+
+    for (Broker aliveBroker : clusterModel.aliveBrokers()) {
+      if (aliveBroker.leaderReplicas().size() > _balanceLowerLimit) {
+        eligibleBrokers.add(aliveBroker);
+      }
+    }
+    List<Broker> candidateBrokers = Collections.singletonList(broker);
+    Set<String> excludedTopics = optimizationOptions.excludedTopics();
+    while (!eligibleBrokers.isEmpty()) {
+      Broker sourceBroker = eligibleBrokers.poll();
+      for (Replica replica : new HashSet<>(sourceBroker.leaderReplicas())) {
+        if (shouldExclude(replica, excludedTopics) || broker.replica(replica.topicPartition()) != null) {
+          continue;
+        }
+        Broker b = maybeApplyBalancingAction(clusterModel,
+                                             replica,
+                                             candidateBrokers,
+                                             ActionType.INTER_BROKER_REPLICA_MOVEMENT,
+                                             optimizedGoals,
+                                             optimizationOptions);
+        // Only need to check status if the action is taken. This will also handle the case that the source broker
+        // has nothing to move in. In that case we will never reenqueue that source broker.
+        if (b != null) {
+          if (broker.leaderReplicas().size() >=  _balanceLowerLimit) {
+            return false;
+          }
+          // If the source broker has a lower number of leader replicas than the next broker in the eligible broker
+          // queue, we reenqueue the source broker and switch to the next broker.
+          if (!eligibleBrokers.isEmpty() && sourceBroker.leaderReplicas().size() < eligibleBrokers.peek().leaderReplicas().size()) {
+            eligibleBrokers.add(sourceBroker);
+            break;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  private void moveReplicasOffDeadBroker(Broker broker,
+                                         ClusterModel clusterModel,
+                                         Set<Goal> optimizedGoals,
+                                         OptimizationOptions optimizationOptions) {
+    // Get the eligible brokers.
+    SortedSet<Broker> candidateBrokers = new TreeSet<>(Comparator.comparingInt((Broker b) -> b.replicas().size())
+                                                                 .thenComparingInt(Broker::id));
+    candidateBrokers.addAll(clusterModel.aliveBrokers());
+
+    Set<String> excludedTopics = optimizationOptions.excludedTopics();
```

Comment:
If we intend to use this function for removing replicas off of a dead broker, why would we check `excludedTopics`?